src/games/strategy/triplea/settings/ValueRange.java

```diff
@@ -0,0 +1,23 @@
+package games.strategy.triplea.settings;
+
+/**
+ * Encapsulates the permissible range for the value of a specific setting.
+ *
+ * <p>
+ * Implementations may choose to track other associated values in addition to the range itself, such as a default value.
+ * </p>
+ */
+public interface ValueRange {
```

Comment:
@DanVanAtta Agreed.  A _Null Object_ (for either `String` or `ValueRange`) seems cleaner than an `Optional` in this case.

Can you elaborate on how you see a `Supplier` replacing the new `getValueRangeXXX()` method on `SettingInputComponent`?  Just for clarification, `SettingInputComponent` has seven existing methods--it was not a new interface (with a single method) that I added as part of this PR.